### PR TITLE
[BEAM-1347] Update protos related to State API for prototyping purposes

### PR DESCRIPTION
**Pull request by lukecwik**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by lukecwik** [comment]
R: @tgroh 

None of the State API protos are currently used or instantiated. For protos that are used, only fields were added.

File: sdks/common/fn-api/src/main/proto/beam_fn_api.proto [hunk 1]
```diff
@@ -540,91 +536,82 @@ service BeamFnState {
...
-  // One of the two fields below are required to be set.
-  // If state key is set, then the State API should be invoked to fetch the
-  // values allowing one to restart the iterable. Otherwise the bytes for the
-  // entire iterable are represented and should be decoded using an iterable
-  // coder using the outer context.
-  StateKey state_key = 1;
-  repeated bytes iterable = 2;
+// A logical byte stream which can be continued using the state API.
+message ContinuableStream {
+  // An empty continuation token represents the end of the logical byte stream.
```
Comment (tgroh) [inline sdks/common/fn-api/src/main/proto/beam_fn_api.proto]: Required? (but may be empty)

Comment (lukecwik) [inline sdks/common/fn-api/src/main/proto/beam_fn_api.proto]: It turns out that I meant to say that unsef = EOF, and not empty byte array = EOF

File: sdks/common/fn-api/src/main/proto/beam_fn_api.proto [hunk 2]
```diff
@@ -465,34 +471,33 @@ service BeamFnData {
...
   string instruction_reference = 2;
 
-  // At least one of the following fields should be populated.
-  // Also, no request should use a state key referred to in another state key.
+  // (Required) The state key this request is for.
+  StateKey state_key = 3;
 
-  // (Optional) A request to get state.
-  repeated StateGetRequest get = 3;
+  // At least one of the following fields must be populated.
```
Comment (lukecwik) [inline sdks/common/fn-api/src/main/proto/beam_fn_api.proto]: Done

File: sdks/common/fn-api/src/main/proto/beam_fn_api.proto [hunk 3]
```diff
@@ -570,7 +570,9 @@ message StateKey {
 
 // A logical byte stream which can be continued using the state API.
 message ContinuableStream {
-  // An empty continuation token represents the end of the logical byte stream.
+  // If specified, represents a token which can be used with the state API
```
Comment (tgroh) [inline sdks/common/fn-api/src/main/proto/beam_fn_api.proto]: required/optional?

File: sdks/common/fn-api/src/main/proto/beam_fn_api.proto [hunk 4]
```diff
@@ -570,7 +570,9 @@ message StateKey {
...
+  // to get the next chunk of this logical byte stream. The end of the logical
+  // byte stream is signalled by this field being unset.
   bytes continuation_token = 1;
 
   // Represents a part of a logical byte stream. Elements within
```
Comment (tgroh) [inline sdks/common/fn-api/src/main/proto/beam_fn_api.proto]: required/optional?

Comment (lukecwik) [inline sdks/common/fn-api/src/main/proto/beam_fn_api.proto]: Done

**Pull request closed without merging**